crategen/validators.py

```diff
@@ -0,0 +1,109 @@
+from pydantic import ValidationError
+from typing import Union
+from .models import WRROCProcess, WRROCWorkflow, WRROCProvenance
+from urllib.parse import urlparse
+
+def validate_wrroc(data: dict) -> Union[WRROCProvenance, WRROCWorkflow, WRROCProcess]:
+    """
+    Validate that the input data is a valid WRROC entity and determine which profile it adheres to.
+    
+    This function attempts to validate the input data against the WRROCProvenance model first.
+    If that validation fails, it attempts validation against the WRROCWorkflow model.
+    If that also fails, it finally attempts validation against the WRROCProcess model.
+    
+    Args:
+        data (dict): The input data to validate.
+    
+    Returns:
+        Union[WRROCProvenance, WRROCWorkflow, WRROCProcess]: The validated WRROC data, indicating the highest profile the data adheres to.
+    
+    Raises:
+        ValueError: If the data does not adhere to any of the WRROC profiles.
+    """
+    # Convert '@id' to 'id' for validation purposes
+    if '@id' in data:
+        data['id'] = data.pop('@id')
```

Comment:
Why is this necessary? Is `@id` not an allowed field name in Pydantic? If so, did you try setting [aliases](https://docs.pydantic.dev/latest/concepts/alias/#aliaspath-and-aliaschoices)?

Reply:
I've replaced the manual renaming of `@id` with Pydantic’s alias functionality. This makes the code more maintainable and consistent with Pydantic’s capabilities.

Reply:
Where in the models are you actually using `@id` aliases now? I didn't look at the models in depth, but I didn't see them (or any import of `Fields` or the use of `alias`). Rather, it looks like the models still just use `id`? Again, maybe I'm confused, so a pointer would help.